src/javaharness/java/arcs/api/DeviceClientImpl.java

```diff
@@ -47,11 +52,7 @@ public void receive(String json) {
     switch (message) {
       case MESSAGE_READY:
         logger.info("logger: Received 'ready' message");
-        // TODO: Onstartup Arcs should be configuration based, not hardcoded.
-        environment.sendMessageToArcs(jsonParser.stringify(jsonParser.emptyObject()
-            .put("message", "runArc")
-            .put("recipe", "Ingestion")
-            .put("arcid", "ingestion-arc")), null);
+        readyListeners.forEach(listener -> listener.onReady(content.getArray(FIELD_READY_RECIPES).asStringArray()));
```

Comment:
You could probably avoid the need to inject readylisteners by just having a package private method on the ShellApiArcsEnvironment like environment.fireReadyEvent(...)

Reply:
thanks for the suggestion, done.